cli/tx_commit.go

```diff
@@ -15,22 +15,23 @@ import (
 
 	"github.com/spf13/cobra"
 
-	"github.com/sourcenetwork/defradb/config"
 	"github.com/sourcenetwork/defradb/http"
 )
 
-func MakeTxCommitCommand(cfg *config.Config) *cobra.Command {
+func MakeTxCommitCommand() *cobra.Command {
 	var cmd = &cobra.Command{
 		Use:   "commit [id]",
 		Short: "Commit a DefraDB transaction.",
 		Long:  `Commit a DefraDB transaction.`,
 		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			cfg := mustGetContextConfig(cmd)
+
 			id, err := strconv.ParseUint(args[0], 10, 64)
 			if err != nil {
 				return err
 			}
-			tx, err := http.NewTransaction(cfg.API.Address, id)
+			tx, err := http.NewTransaction(cfg.GetString("api.address"), id)
```

Comment:
question: This feels like it should be a shared const somewhere, not hardcoded here - do you have reasons for declaring it here like this?

(same for the other locations where this is done)

Reply:
I wanted to avoid having const keys or a pre-defined config struct so that the config can be modified more easily. The places that access the `api.address` specifically should have good test coverage since they use random port assignments in the tests.